tests/collections/asr/decoding/test_fast_rnnt_greedy_decoding.py

```diff
@@ -0,0 +1,187 @@
+import glob
+import json
+import os
+import sys
+import tempfile
+import traceback
+
+import ipdb
+import jiwer
+import pytest
+import torch
+from omegaconf import OmegaConf, open_dict
+
+from nemo.collections.asr.models import ASRModel
+from nemo.collections.asr.models.rnnt_bpe_models import EncDecRNNTBPEModel
+from nemo.collections.asr.parts.submodules.fast_rnnt_greedy_decoding import RNNTGreedyDecodeFast
+from nemo.collections.asr.parts.submodules.rnnt_greedy_decoding import GreedyBatchedRNNTInfer
+
+
+@pytest.mark.parametrize(
+    ("model_name", "batch_size", "use_subset", "enable_bfloat16"),
+    [
+        # The xfail is not catching for some reason...
+        # pytest.param("stt_en_fastconformer_transducer_large", 8, True, True, marks=pytest.mark.xfail(reason="Cannot instantiate graph with persistent RNN")),
+        ("stt_en_fastconformer_transducer_large", 7, False, True),
+        ("stt_en_fastconformer_transducer_xlarge", 16, False, True),
+        ("stt_en_fastconformer_transducer_xlarge", 16, True, True),
+        ("stt_en_fastconformer_transducer_xlarge", 128, True, True),
+        ("stt_en_fastconformer_transducer_xlarge", 16, False, False),
+    ],
+)
+def test_for_loop(model_name, batch_size, use_subset, enable_bfloat16):
+    nemo_model = ASRModel.from_pretrained(model_name, map_location="cuda")
+    conf = nemo_model.to_config_dict()
+    with open_dict(conf):
+        conf["decoding"]["greedy"]["max_symbols"] = 5
+        conf["decoding"]["greedy"]["loop_labels"] = False
+
+    with tempfile.NamedTemporaryFile() as fp:
```

Comment:
`change_decoding_strategy` should be enough. Saving/restoring models takes a lot of time, it's better to avoid serialization if possible in unit tests.

Reply:
Thanks for the tip. It sped up test execution almost 3 times.